src/extensions/sb-ncbr/partial-charges/property.ts

```diff
@@ -0,0 +1,192 @@
+import { Model } from '../../../mol-model/structure';
+import { CustomProperty } from '../../../mol-model-props/common/custom-property';
+import { PropertyWrapper } from '../../../mol-model-props/common/wrapper';
+import { ParamDefinition as PD } from '../../../mol-util/param-definition';
+import { MmcifFormat } from '../../../mol-model-formats/structure/mmcif';
+import { CustomPropertyDescriptor } from '../../../mol-model/custom-property';
+import { CustomModelProperty } from '../../../mol-model-props/common/custom-model-property';
+import { arrayMinMax } from '../../../mol-util/array';
+
+type TypeId = number;
+type IdToCharge = Map<number, number>;
+interface ChargesData {
+    typeIdToMethod: Map<TypeId, string>;
+    typeIdToAtomIdToCharge: Map<TypeId, IdToCharge>;
+    typeIdToResidueToCharge: Map<TypeId, IdToCharge>;
+    maxAbsoluteAtomCharges: IdToCharge;
+    maxAbsoluteResidueCharges: IdToCharge;
+    maxAbsoluteAtomChargeAll: number;
+}
+type PartialCharges = PropertyWrapper<ChargesData | undefined>;
+
+const PartialChargesPropertyParams = {
+    typeId: PD.Select<number>(0, [[0, '0']]),
+};
+type PartialChargesPropertyParams = typeof PartialChargesPropertyParams;
+const defaultPartialChargesPropertyParams = PD.clone(PartialChargesPropertyParams);
+
+function getParams(model: Model) {
+    const typeIdToMethod = getTypeIdToMethod(model);
+    const options = Array.from(typeIdToMethod.entries()).map(
+        ([typeId, method]) => [typeId, method] as [number, string]
+    );
+    return {
+        typeId: PD.Select<number>(1, options),
+    };
+}
+
+async function getData(model: Model): Promise<CustomProperty.Data<PartialCharges>> {
+    const info = PropertyWrapper.createInfo();
+
+    if (!SbNcbrPartialChargesPropertyProvider.isApplicable(model)) return { value: { info, data: undefined } };
+
+    const typeIdToMethod = getTypeIdToMethod(model);
+    const typeIdToAtomIdToCharge = getTypeIdToAtomIdToCharge(model);
+    const typeIdToResidueToCharge = getTypeIdToResidueIdToCharge(model, typeIdToAtomIdToCharge);
+    const maxAbsoluteAtomCharges = getMaxAbsoluteCharges(typeIdToAtomIdToCharge);
+    const maxAbsoluteResidueCharges = getMaxAbsoluteCharges(typeIdToResidueToCharge);
+    const maxAbsoluteAtomChargeAll = getMaxAbsoluteAtomChargeAll(maxAbsoluteAtomCharges, maxAbsoluteResidueCharges);
+
+    return {
+        value: {
+            info,
+            data: {
+                typeIdToMethod,
+                typeIdToAtomIdToCharge,
+                typeIdToResidueToCharge,
+                maxAbsoluteAtomCharges,
+                maxAbsoluteResidueCharges,
+                maxAbsoluteAtomChargeAll,
+            },
+        },
+    };
+}
```

Comment:
I think the `PropertyWrapper` around `ChargesData` is redundant and you should be able to return just `CustomProperty.Data<ChargesData | undefined>` here and modify the rest of the code accordingly.

I know it's used in the other extension which I pointed you to, but no need to use that here as the `PropertyWrapper.createInfo();` isn't used anywhere relevant to this extension (I hope).

-----

Can you try something like this and modifying the usage of this code?

```suggestion
async function getData(model: Model): Promise<CustomProperty.Data<ChargesData | undefined>> {
    if (!SbNcbrPartialChargesPropertyProvider.isApplicable(model)) return undefined;

    const typeIdToMethod = getTypeIdToMethod(model);
    const typeIdToAtomIdToCharge = getTypeIdToAtomIdToCharge(model);
    const typeIdToResidueToCharge = getTypeIdToResidueIdToCharge(model, typeIdToAtomIdToCharge);
    const maxAbsoluteAtomCharges = getMaxAbsoluteCharges(typeIdToAtomIdToCharge);
    const maxAbsoluteResidueCharges = getMaxAbsoluteCharges(typeIdToResidueToCharge);
    const maxAbsoluteAtomChargeAll = getMaxAbsoluteAtomChargeAll(maxAbsoluteAtomCharges, maxAbsoluteResidueCharges);

    return {
        typeIdToMethod,
        typeIdToAtomIdToCharge,
        typeIdToResidueToCharge,
        maxAbsoluteAtomCharges,
        maxAbsoluteResidueCharges,
        maxAbsoluteAtomChargeAll,
    };
}
```

Reply:
Yes, the `info` property is never used.